posts/Yoann/2019-11-08-hibernate-search-quarkus.adoc

```diff
@@ -0,0 +1,1428 @@
+It integrates many Java libraries and provides tooling to ease development.
+Quarkus-based applications can run in a traditional HotSpot setup,
+but also (via https://www.graalvm.org/[GraalVM])
+as native binaries with a tiny memory footprint and impressively fast boot time.
+
+The latter is what really sets Quarkus apart:
+it means that Quarkus-based applications are especially well-suited for running in a container.
+
+=== What's Hibernate Search?
+
+http://hibernate.org/search/[Hibernate Search] is a Java library that augments traditional ORM-based applications
+with advanced search features, including in particular full-text search.
+It integrates into http://hibernate.org/orm/[Hibernate ORM]
+to automatically index entities into https://www.elastic.co/products/elasticsearch[Elasticsearch],
+and provides a Java API to seamlessly search for entities based on Elasticsearch indexes.
+
+== Goal
+
+In this blog post, we will explore how to write a REST service responsible for managing a list of clients
+and their assigned business manager.
+
+We will use the https://quarkus.io/[Quarkus] framework to quickly and easily write a container-ready application,
+http://hibernate.org/orm/[Hibernate ORM] to handle persistence to a database,
+and http://hibernate.org/search/[Hibernate Search] to add full-text search features to the application.
+
+The application will be backed by https://www.postgresql.org/[PostgreSQL] to store normalized, relational data,
+and https://www.elastic.co/products/elasticsearch[Elasticsearch] to index part of this data in a de-normalized,
+easy-to-search fashion.
+
+== Code
+
+You can find the code of the resulting REST application
+in the `hibernate/hibernate-demos` repository on GitHub:
+link:{demoRepoUrl}[].
+
+== Prerequisites
+
+The application described in the following sections was built using the following tools:
+
+* A shell
+* OpenJDK 8 (OpenJDK 13 works just as well).
+You can get binaries from https://www.oracle.com/technetwork/java/javase/downloads/index.html[Oracle],
+or from the https://adoptopenjdk.net/[AdoptOpenJDK project], among others.
+* https://maven.apache.org/download.cgi[Apache Maven] 3.6.2.
+* https://www.docker.com/[Docker] and in particular https://docs.docker.com/compose/[Docker-compose]
+to run PostgreSQL and Elasticsearch.
+
+GraalVM is necessary in order to build a native image of the application;
+see https://quarkus.io/get-started/ for details.
+
+== Initial CRUD application
+
+Before we add full-text search to our application,
+we need some data to work with.
+This section will explain how to create a basic CRUD application with Quarkus.
+
+If you are already familiar with this, just skip through this section and go right to <<adding-hibernate-search>>.
+
+=== Initializing the project
+
+Quarkus provides a Maven plugin to initialize the layout of a new project.
+Just run this:
+
+[source]
+----
+$ mvn io.quarkus:quarkus-maven-plugin:0.28.0:create \
```

Comment:
Let's use 1.0.0.CR1?

Reply:
I did, and updated all the examples. I just forgot to update this one.